modules/openthread/platform/platform-zephyr.h

```diff
@@ -17,6 +18,7 @@
 
 #include <openthread/instance.h>
 #include <zephyr/net/net_pkt.h>
+#include <zephyr/net/openthread.h>
```

Comment:
Is this actually needed, I don't see any usage of the contents of this header within `platform-zephyr.h`, and it is the actual  culprit for the test breakages we see (now it's about undefined symbols).